invenio_drafts_resources/services/records/components/files_base.py

```diff
@@ -0,0 +1,239 @@
+                )
+                return  # exit early
+
+        files.enabled = enabled
+
+    def update_draft(self, identity, data=None, record=None, errors=None):
+        """Assigns files.enabled and warns if files are missing.
+
+        NOTE: `record` actually refers to the draft
+              (this interface is used in records-resources and rdm-records)
+        """
+        draft = record
+        draft_files = self.get_record_files(draft)
+        enabled = data.get(self.files_data_key, {}).get("enabled", True)
+        default_preview = data.get(self.files_data_key, {}).get("default_preview")
+        if draft_files.enabled != enabled:
+            if not self.service.check_permission(
+                identity, "manage_files", record=draft
+            ):
+                errors.append(
+                    {
+                        "field": f"{self.files_data_key}.enabled",
+                        "messages": [
+                            _("You don't have permissions to manage files options.")
+                        ],
+                    }
+                )
+                return  # exit early
+
+        try:
+            self.files_component.assign_files_enabled(draft, enabled)
+        except ValidationError as e:
+            errors.append(
+                {"field": f"{self.files_data_key}.enabled", "messages": e.messages}
+            )
+            return  # exit early
+
+        if draft_files.enabled and not draft_files.items():
+            errors.append(
+                {
+                    "field": f"{self.files_data_key}.enabled",
+                    "messages": [
+                        _(
+                            "Missing uploaded files. To disable files for "
+                            "this record please mark it as metadata-only."
+                        )
+                    ],
+                }
+            )
+
+        try:
+            self.files_component.assign_files_default_preview(
+                draft,
+                default_preview,
+            )
+        except ValidationError as e:
+            errors.append(
+                {
+                    "field": f"f{self.files_data_key}.default_preview",
+                    "messages": e.messages,
+                }
+            )
+
+    def edit(self, identity, draft=None, record=None):
+        """Edit callback."""
+        draft_files = self.get_record_files(draft)
+        record_files = self.get_record_files(record)
+        draft_bucket = self.get_record_bucket(draft)
+        if draft_bucket is None:
+            # Happens, when a soft-deleted draft is un-deleted.
+            draft[self.files_data_key] = {"enabled": True}
+            # re-fetch the files attribute - above data getter sets the attribute
+            draft_files = self.get_record_files(draft)
+            draft_files.create_bucket()
+        draft_files.copy(record_files)
+        # force going through the new version
+        draft_files.lock()
```

Comment:
this part was missing in the regular files - the bucket was not locked, meaning you could edit files in the draft, but they were not copied over to the record - because we require new version for the  regular files changes - resulting in some zombie draft files